src/packet/compressed_data.js

```diff
@@ -205,8 +209,12 @@ function zlib(compressionStreamInstantiator, ZlibStreamedConstructor) {
 
 function bzip2Decompress() {
   return async function(data) {
-    const { decode: bunzipDecode } = await import('@openpgp/seek-bzip');
-    return streamFromAsync(async () => bunzipDecode(await streamReadToEnd(data)));
+    const { default: Unbzip2Stream } = await import('@openpgp/unbzip2-stream');
+    if (isArrayStream(data)) {
+      data = await streamReadToEnd(data);
+    }
+    data = toStream(data);
+    return data.pipeThrough(new Unbzip2Stream());
```

Comment:
This is more of a comment on the stream tools API, but I think the interaction between the helpers is confusing; I think we should have `ArrayStream` support pipeThrough; or alternatively add a (required) arg to `toStream` like `convertArrayStream` so that it takes care of it internally.

Reply:
`ArrayStream` does support `pipeThrough`, however, in that case all of the data gets decompressed regardless of what happens after it, since `ArrayStream`s don't have backpressure support.

In this case, since we want to limit the output size of the stream afterwards (in a future PR), we want to use a real stream.

You're right that the docs need updating, though.